R/vertica.udx.util.R

```diff
@@ -0,0 +1,92 @@
+#x is a data.frame
+#y is a list of which the first and only element is a function to run
+
+
+possible_output_types = c("boolean","int","float","real","char","varchar","longvarchar","date","datetime","smalldatetime","time","timestamp","timestamptz","timetz","numeric","varbinary","longvarbinary","binary")
+
+vertica_R_mapping <- function(type)
+{
+	if(type == "double")
+		return ("float")
+	if(type == "integer")
+		return ("int")
+	if(type == "character")
+		return ("varchar")
+}
+
+
+function_charsize = 30*1000*1000
+
+
+### the main udx execution fuction that does the work
+### receives 2 parameters params and outcols
+### params are parameters for the function
+### outcols are the output column types of the function
+
+vertica_function_executor <- function(x,y)
+{
+	parameters = deserialize_object(y$params)
+
+	#extract the function
+	vertica.dplyr.func = parameters[["vertica.dplyr.func"]]
+	parameters["vertica.dplyr.func"] <- NULL
+
+	#extract and load the necessary libraries
+	libraries = parameters[["vertica.dplyr.libraries"]]
+	parameters["vertica.dplyr.libraries"] <- NULL
+	if(length(libraries) > 0)
+		sapply(libraries, function(x) library(x,character.only = TRUE))
+
+	list2env(parameters, environment(vertica.dplyr.func))
+	
+     	output = vertica.dplyr.func(x)
+	return(output)
+}
+
+
+vertica_function_executor.OutputCallback <- function(x,y)
+{
+	outcols = deserialize_object(y$outcols)
```

Comment:
Since you are just reading outcols here, and you set outcols to values you get from test_fun, can you just port test_fun's logic to here? Since you are anyway running the function twice, would it be more efficient to just run the dummy invocation in here, saving an additional round-trip invocation to your function_executor UDx? I guess it depends on when/how many times Vertica calls OutputCallback.